implemented

u11107 opened the pull request:
-adding to the library

Commit:
implemented
-adding to the library

Charmandik commented on app/src/main/java/com/example/playlistmaker/media/ui/fragment/FavouriteTracksFragment.kt:
```diff
@@ -0,0 +1,80 @@
+            is FavouriteTracksState.Content -> {
+                favoritesTracksAdapter.clearTracks()
+                favoritesTracksAdapter.tracks = state.tracks as MutableList<Track>
+                binding.favouriteTracksRecycler.visibility = View.VISIBLE
```
Для управления видимостью можно использовать свойство isVisible, устанавливая значение false для View.GONE и true для View.VISIBLE
Например, если нам нужно установить View.VISIBLE:
myViewName.isVisible = true
Или если нужно установить View.GONE, то : 
myViewName.isVisible = false

Возможно, тебе понравится такой подход

Charmandik commented on app/src/main/java/com/example/playlistmaker/util/App.kt:
```diff
@@ -18,10 +19,7 @@ import java.util.Locale
     companion object {
-        const val EXAMPLE_PREFERENCES = "example_preferences"
-        const val KEY_THEME = "key"
         lateinit var sharedMemory: SharedPreferences
```
Вместо использования lateinit, я рекомендую тебе использовать подход с добавлением nullability к типу объекта с оператором ?. и сразу проинициализировать его значением null. Это решение позволяет избежать ошибок, связанных с попытками доступа к неинициализированным объектам, которые могут привести к непредсказуемому поведению или крашам приложения. Хотя это может добавить некоторую дополнительную работу при обращении к объекту через оператор ?. , но это значительно повысит безопасность приложения, так как ты будешь уверен, что объект всегда либо имеет значение, либо равен null, и не будет вызывать исключительных ситуаций.Можешь еще прочитать статью на эту тему https://habr.com/ru/companies/maxilect/articles/590061

Charmandik commented on app/src/main/java/com/example/playlistmaker/media/data/FavouriteTracksRepositoryImpl.kt:
```diff
@@ -0,0 +1,37 @@
+        emit(convertFromTrackEntity(tracks))
+    }
+
+    override suspend fun isFavoriteTrack(trackId: Int): Flow<Boolean> = flow {
```
В Kotlin лучше использовать "return" вместо "=", потому что это помогает делать код более читаемым и явным. Когда ты используешь "return", это явно указывает, что функция возвращает определенное значение. В то же время, использование "=" может вызывать путаницу, особенно если в функции есть несколько выражений.

Charmandik commented on app/src/main/res/layout/activity_audio_player.xml:
```diff
@@ -173,12 +186,13 @@
-        app:layout_constraintTop_toTopOf="parent"
-        app:srcCompat="@drawable/arrow_back"
-        tools:ignore="ContentDescription" />
+<!--    <ImageView-->
```
Не рекомендую оставлять закомментированный код т. к как правило он долго не удаляется и визуально загрязнаяет код

Charmandik commented on app/src/main/java/com/example/playlistmaker/media/ui/viewModel/FavouriteTracksViewModel.kt:
```diff
@@ -0,0 +1,50 @@
+) : ViewModel() {
+
+    private var isClickAllowed = true
+    private val stateLiveData = MutableLiveData<FavouriteTracksState>()
```
Для ситуаций, когда нам нужно иметь два состояния( мутабельное и иммутабельное), рекомендую использовать подход с добавлением префикса в виде нижнего подчеркивания для мутабельного состояния(которое должно быть приватным):
private val _state = MutableLiveData<MyState>()
val state : LiveData<MyState> = _state

Так ты сможешь не придумывать разные имена, чтобы не повторяться, но и при этом нормально передавать смысл и сможешь точно понимать, что эта вот переменная это просто копия той, но иммутабельна

Commit:
corrected
- Button to add to favorite tracks in the night theme
-Moved classes to the data package
